test/mix/tasks/phx.gen.schema_test.exs

```diff
@@ -36,6 +36,98 @@ defmodule Mix.Tasks.Phx.Gen.SchemaTest do
     end
   end
 
+  test "generates custom table name", config do
+    in_tmp_project config.test, fn ->
+      Gen.Schema.run(~w(Blog.Post posts --table cms_posts))
+
+      assert [migration] = Path.wildcard("priv/repo/migrations/*_create_blog_post.exs")
+
+      assert_file migration, fn file ->
+        assert file =~ "create table(:cms_posts) do"
+      end
+    end
+  end
+
+  test "generates nested context", config do
+    in_tmp_project config.test, fn ->
+      Gen.Schema.run(~w(Blog.Admin.User users name:string))
+
+      assert [migration] = Path.wildcard("priv/repo/migrations/*_create_blog_admin_user.exs")
+
+      assert_file migration, fn file ->
+        assert file =~ "defmodule Phoenix.Repo.Migrations.CreatePhoenix.Blog.Admin.User do"
+        assert file =~ "create table(:users) do"
+      end
+
+      assert_file "lib/phoenix/blog/admin/user.ex", fn file ->
+        assert file =~ "defmodule Phoenix.Blog.Admin.User do"
+        assert file =~ "schema \"users\" do"
+      end
+    end
+  end
+
+  test "generates belongs_to associations with association table provided by user", config do
+    in_tmp_project config.test, fn ->
+      Gen.Schema.run(~w(Blog.Post posts title user_id:references:users))
+
+      assert [migration] = Path.wildcard("priv/repo/migrations/*_create_blog_post.exs")
+
+      assert_file migration, fn file ->
+        assert file =~ "add :user_id, {:references, :users}"
```

Comment:
My bad, I'll fix it. But, shouldn't the belongs_to be in schema instead of migration?